### DISCUSSION TITLE
Register a dynamic URL with

### DISCUSSION THREAD

**Discussion by helmutkaufmann** [wintercms/winter#1023] [Q&A] [answered]
I have a plugin with a a component component called ``Admin``.  This component is added normally on a CMS page ``{% component 'Admin' %}``. The component has a parameter ``viewerURL``. I would like to register a URL that takes the value of ``viewerURL`` and registers it as an URL. For example, if the viewerURL is ``xyz`` it should register ``https://myserver.com/xyz``. The only content on that page should be ``{% component 'View' %}`` with the default layout.

At the moment, I create those CMS pages manually and it somehow seems to be pointless as the content returned is always ``{% component 'View' %}``

**Comment by helmutkaufmann** [comment]
Perhaps the question is simpler: In plugin, can I define a route that renders a CMS page that I define within the plugin, i.e. I do not want to manually create the CMS in the backend but have it defined by the plugin.

**Comment by WebVPF** [comment]
You can define these routes by creating a file named `routes.php` in a same directory as the plugin registration.
Docs https://wintercms.com/docs/v1.2/docs/services/router

Views are typically stored in the `views` directory of a plugin.
Docs:
- https://wintercms.com/docs/v1.2/docs/services/response-view#view-responses
- https://wintercms.com/docs/v1.2/docs/services/response-view#views

Examples:
- https://github.com/WebVPF/wn-robots-plugin
- https://github.com/WebVPF/wn-trafficadvice-plugin
- https://www.youtube.com/watch?v=HED4xxWA9-c&list=PLUBR53Dw-Ef-X-_A1KLy41r2QArCBy4rM&index=38

**Reply by helmutkaufmann** [reply]
Thanks a lot.

**Reply by helmutkaufmann** [reply]
Ok, the view part I do understand. One last bit that is missing: I would like to render the page that comes from the view using the default layout. If you would not mind pointing me into the right direction, that would be great. Thank you.

**Reply by WebVPF** [reply, marked as answer]
Then you'd better use CMS Pages.
